plugins/postgres/dbt/adapters/postgres/connections.py

```diff
@@ -140,6 +141,11 @@ def connect():
                 connect_timeout=credentials.connect_timeout,
                 **kwargs,
             )
+
+            # Register extra data types to avoid dbt contract errors
+            register_uuid()
+            register_ipaddress()
```

Comment:
If I understand right -

This change will allow users to specify `UUID`, `CIDR`, `INET`, etc as data types within model contracts. These are standard types that were added in later versions of Postgres / psycopg2, which is why they're opt-in here. Is there any risk / downside to registering them here?

This still won't support [totally custom data types](https://www.postgresql.org/docs/current/sql-createtype.html) that users may define within their Postgres instances.

Reply:
Yeap! **UUID** + **IP** stuff probably will cover most cases!

Not sure if `dbt` should handle edge or user specific cases right now (like custom data types) since contracts are not widespread and this will need a plugin-like functionality on the adapter to extend the range of accepted types.